src/base/bittorrent/session.cpp

```diff
@@ -920,6 +928,75 @@ void Session::adjustLimits(libt::settings_pack &settingsPack)
                          , maxActive > -1 ? maxActive + m_extraLimit : maxActive);
 }
 
+void Session::initMetrics()
+{
+    m_metricIndicies.net.hasIncomingConnections = libt::find_metric_idx("net.has_incoming_connections");
```

Comment:
You are mistaken here. Let's compare these two algorithms:
(using `find_metric_idx`)
```
foreach neededMetrics
    foreach allMetrics
        if neededMetricName == metricName
            neededMetricIndex = metricIndex
```
(using `session_stats_metrcs`)
```
foreach allMetrics
    foreach neededMetrics
        if neededMetricName == metricName
            neededMetricIndex = metricIndex
```